services/migrations/gitlab.go

```diff
@@ -640,13 +640,22 @@ func (g *GitlabDownloader) GetReviews(context base.IssueContext) ([]*base.Review
 		return nil, err
 	}
 
+	var createdAt time.Time
+	if approvals.UpdatedAt != nil {
+		createdAt = *approvals.UpdatedAt
+	} else if approvals.CreatedAt != nil {
+		createdAt = *approvals.CreatedAt
+	} else {
+		createdAt = time.Now()
+	}
+
```

Comment:
Why did the migrator use `approval.UpdatedAt` for `review.CreatedAt`?

It seems more reasonable to use `review.CreatedAt = approvals.CreatedAt ? approvals.CreatedAt : zero` ....

(Although both are fine to me, I just want to make it clear why the logic looks like this ....)

Reply:
I'm als for first check if CreatedAt != nil and fall back to UpdatedAt next. if we need Update too - we should add it to the base.Review - but that's nothing for a bugfix

Reply:
@realaravinth

Reply:
> Why did the migrator use `approval.UpdatedAt` for `review.CreatedAt`?
> 
> It seems more reasonable to use `review.CreatedAt = approvals.CreatedAt ? approvals.CreatedAt : zero` ....
> 
> (Although both are fine to me, I just want to make it clear why the logic looks like this ....)


I agree that it would make more sense to use `approval.CreatedAt` and I cannot speak to the motives of the original implementor. I preserved this logic to not introduce a semantic change in the bugfix, however small. That being said, if you think this should be added to this patch, I'd be happy to do it.

---

[source](https://lab.fedeproxy.eu/forgefriends/forgefriends/-/merge_requests/37#note_915)

Reply:
If we all think that `approval.CreatedAt` is more reasonable, then we can have it now (we are making things better!).

It's hard to say whether the legacy code is correct or not 😂